app/src/lib/shells/win32.ts

```diff
@@ -43,75 +45,127 @@ export async function getAvailableShells(): Promise<
     },
   ]
 
+  const powerShellPath = await findPowerShell()
+  if (powerShellPath != null) {
+    shells.push({
+      shell: Shell.PowerShell,
+      path: powerShellPath,
+    })
+  }
+
+  const hyperPath = await findHyper()
+  if (hyperPath != null) {
+    shells.push({
+      shell: Shell.Hyper,
+      path: hyperPath,
+    })
+  }
+
+  const gitBashPath = await findGitBash()
+  if (gitBashPath != null) {
+    shells.push({
+      shell: Shell.GitBash,
+      path: gitBashPath,
+    })
+  }
+
+  return shells
+}
+
+async function findPowerShell(): Promise<string | null> {
   const powerShell = enumerateValues(
     HKEY.HKEY_LOCAL_MACHINE,
     'Software\\Microsoft\\Windows\\CurrentVersion\\App Paths\\PowerShell.exe'
   )
-  if (powerShell.length > 0) {
-    const first = powerShell[0]
-
-    // NOTE:
-    // on Windows 7 these are both REG_SZ, which technically isn't supposed
-    // to contain unexpanded references to environment variables. But given
-    // it's also %SystemRoot% and we do the expanding here I think this is
-    // a fine workaround to do to support the maximum number of setups.
-
-    if (
-      first.type === RegistryValueType.REG_EXPAND_SZ ||
-      first.type === RegistryValueType.REG_SZ
-    ) {
-      const path = first.data.replace(
-        /^%SystemRoot%/i,
-        process.env.SystemRoot || 'C:\\Windows'
+
+  if (powerShell.length === 0) {
+    return null
+  }
+
+  const first = powerShell[0]
+
+  // NOTE:
+  // on Windows 7 these are both REG_SZ, which technically isn't supposed
+  // to contain unexpanded references to environment variables. But given
+  // it's also %SystemRoot% and we do the expanding here I think this is
+  // a fine workaround to do to support the maximum number of setups.
+
+  if (
+    first.type === RegistryValueType.REG_EXPAND_SZ ||
+    first.type === RegistryValueType.REG_SZ
+  ) {
+    const path = first.data.replace(
+      /^%SystemRoot%/i,
+      process.env.SystemRoot || 'C:\\Windows'
+    )
+
+    if (await pathExists(path)) {
```

Comment:
The cumulative effect of checking the path exists is a couple of milliseconds (`master` takes `~26ms`, this branch takes `~31ms`) and given this is deferred until you first open the Options/Preferences dialog - and is cached - I'm fine with being more safe here.